### PR TITLE
Translate mapeditor

### PR DESCRIPTION
**Pull request by IvanSavenko**
Another subtask of #1300 
Implemented translation of second Qt-based project - map editor.

Changes:
- [x] Implement translations for Map Editor using Qt translator
- [x] Marked all non-translatable strings in Map Editor UI as such
- [x] Added Ukrainian translation
- [x] German translation by Laserlicht
- [x] Polish translation by dydzio

Language selection relies on selection made in Launcher, or on manual edit of config file.
Unlike Launcher, where language can be changed in runtime, Map Editor loads translation once on startup

### CODE REVIEW

**Comment by IvanSavenko** [comment]
@dydzio0614 
@Laserlicht
If/when you are willing to translate map editor, feel free to do so & send me resulting .ts file.
Blank files available here:
https://github.com/IvanSavenko/vcmi/tree/translate_mapeditor/mapeditor/translation

**Comment by Laserlicht** [comment]
> @dydzio0614 @Laserlicht If/when you are willing to translate map editor, feel free to do so & send me resulting .ts file. Blank files available here: https://github.com/IvanSavenko/vcmi/tree/translate_mapeditor/mapeditor/translation

Of course. 🙂

[german.zip](https://github.com/vcmi/vcmi/files/10305037/german.zip)

File: CMakeLists.txt [hunk 1]
```diff
@@ -327,8 +329,17 @@ endif()
...
 if(ENABLE_LAUNCHER OR ENABLE_EDITOR)
 	# Widgets finds its own dependencies (QtGui and QtCore).
-	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network)
-	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network)
+	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network )
+	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network )
+
+	find_package(QT NAMES Qt6 Qt5 COMPONENTS LinguistTools )
+	find_package(Qt${QT_VERSION_MAJOR} COMPONENTS LinguistTools)
+	if (NOT Qt5LinguistTools_DIR)
```
Comment (Nordsoft91) [inline CMakeLists.txt:337]: no space after if here and everywhere else

File: mapeditor/mainwindow.ui [hunk 2]
```diff
@@ -734,7 +734,7 @@
           <x>0</x>
           <y>0</y>
           <width>128</width>
-          <height>251</height>
+          <height>257</height>
```
Comment (Nordsoft91) [inline mapeditor/mainwindow.ui:737]: why size change is needed?

Comment (kambala-decapitator) [reply mapeditor/mainwindow.ui:737]: that's a standard thing when you open UI file in another OS - it's rendered with native look which can be different in size (e.g. macOS vs KDE)

Comment (IvanSavenko) [reply mapeditor/mainwindow.ui:737]: This ^
The only intended changes in .ui files were:
- marking strings as non-localizeable
- removing some "internal" strings, e.g. FormName, ComboBox fron titles

**Comment by dydzio0614** [comment]
[launcher_pl.ts.txt](https://github.com/vcmi/vcmi/files/10309810/launcher_pl.ts.txt)
[translate.json.txt](https://github.com/vcmi/vcmi/files/10309813/translate.json.txt)
added txt extension so github will accept the files

**Comment by dydzio0614** [comment]
[polish.ts.txt](https://github.com/vcmi/vcmi/files/10310870/polish.ts.txt)
Polish translation with redundant .txt extension to natch github file constraints

File: CI/linux/before_install.sh [hunk 3]
```diff
@@ -5,7 +5,7 @@ sudo apt-get update
 # Dependencies
 sudo apt-get install libboost-all-dev
 sudo apt-get install libsdl2-dev libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev
-sudo apt-get install qtbase5-dev
+sudo apt-get install qtbase5-dev
```
Comment (kambala-decapitator) [inline CI/linux/before_install.sh:8]:
```suggestion
sudo apt-get install qtbase5-dev
```

File: CMakeLists.txt [hunk 4]
```diff
@@ -327,8 +329,17 @@ endif()
 
 if(ENABLE_LAUNCHER OR ENABLE_EDITOR)
 	# Widgets finds its own dependencies (QtGui and QtCore).
-	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network)
-	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network)
+	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network )
+	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network )
```
Comment (kambala-decapitator) [inline CMakeLists.txt:333]:
```suggestion
	find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets Network)
	find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets Network)
```

File: CMakeLists.txt [hunk 5]
```diff
@@ -327,8 +329,17 @@ endif()
...
+
+	find_package(QT NAMES Qt6 Qt5 COMPONENTS LinguistTools )
```
Comment (kambala-decapitator) [inline CMakeLists.txt:335]:
```suggestion
	find_package(QT NAMES Qt6 Qt5 COMPONENTS LinguistTools)
```

File: CMakeLists.txt [hunk 6]
```diff
@@ -327,8 +329,17 @@ endif()
...
+	find_package(Qt${QT_VERSION_MAJOR} COMPONENTS LinguistTools)
+	if(NOT Qt5LinguistTools_DIR)
```
Comment (kambala-decapitator) [inline CMakeLists.txt:337]:
```suggestion
	if(NOT Qt${QT_VERSION_MAJOR}LinguistTools_DIR)
```

File: mapeditor/mainwindow.cpp [hunk 7]
```diff
@@ -115,6 +115,31 @@ void MainWindow::parseCommandLine(ExtractionOptions & extractionOptions)
 			parser.isSet("d")}};
 }
 
+void MainWindow::loadTranslation()
+{
+#ifdef ENABLE_QT_TRANSLATIONS
+	std::string translationFile = settings["general"]["language"].String()+ ".qm";
```
Comment (kambala-decapitator) [inline mapeditor/mainwindow.cpp:121]:
```suggestion
	std::string translationFile = settings["general"]["language"].String() + ".qm";
```

**Review by kambala-decapitator — APPROVED** [review]

**Pull request merged**